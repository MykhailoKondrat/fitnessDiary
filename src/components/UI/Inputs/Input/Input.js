import React from "react";
import classes from "./Input.module.scss";

const Input = (props) => {
  const inputClasses = [classes.Input];
  const inputElement = (
    <>
      <input
        id={props.name}
        type={props.type}
        className={inputClasses.join(" ")}
        onChange={props.change}
        value={props.value}
        placeholder={props.placeholder}
      />
    </>
  );
  let validationError = null;
  if (props.invalid && props.touched) {
    validationError = (
      <p className={classes.errorMessage}>Please enter a valid value!</p>
    );
  }
  return (
    <>
      {inputElement}
      {validationError}
    </>
  );
};

export default Input;
