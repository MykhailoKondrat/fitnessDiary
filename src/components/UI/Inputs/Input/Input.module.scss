@import "src/app/variables.module" ;
.Input{
  width: 80%;
  max-width: 400px;
  margin: 16px 0;
  height: 48px;
  border: 1px solid change-color($mainColor,$saturation:50,$lightness: 80);
  font-size: .8rem;
  outline: none;
  padding: 24px;
  border-radius: 40px;
  &:focus{
    border: 2px solid $mainColor;
    background: change-color($mainColor,$saturation:10,$lightness: 95);
  }
}
.Invalid {
  border-color: indianred;
}
.errorMessage{
  margin-top: -8px;
  font-size: .9rem;
  color: indianred;
  padding-left: 24px;
  width: 80%;
  max-width: 400px;
  text-align: left;
}
