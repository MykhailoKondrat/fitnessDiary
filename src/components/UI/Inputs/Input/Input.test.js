import React from "react";
import {configure,shallow} from 'enzyme';
import { shallowToJson  } from 'enzyme-to-json';
import Adapter from 'enzyme-adapter-react-16';
import renderer from 'react-test-renderer';
import Input from "./Input";


describe('<Input/>', ()=> {
  let wrapper;
  beforeEach( () => {
    wrapper = shallow(<Input />);
  });
  it('should render error paragraph with correct message',  () =>{
    wrapper.setProps({
      valid:false,
      touched:true,
      errorMessage:'Test error message'
    });
    expect(wrapper.find('p.errorMessage')).toHaveLength(1);
    expect(wrapper.find('p.errorMessage').text()).toEqual('Test error message');
  });
 
  it('render correctly Input component', () => {
    // const InputComponent = shallow(<Input valid={false} touched/>);
    expect(shallowToJson(wrapper)).toMatchSnapshot();
  });
})