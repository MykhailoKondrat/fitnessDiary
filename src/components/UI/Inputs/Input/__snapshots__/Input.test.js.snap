// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Input/> render correctly Input component 1`] = `
<Fragment>
  <input
    className="Input"
  />
</Fragment>
`;
